flaml/nlp/huggingface/data_collator.py

```diff
@@ -31,18 +40,23 @@ def __call__(self, features):
         return batch
 
 
-class DataCollatorForPredict(DataCollatorWithPadding):
-    def __call__(self, features):
-        from itertools import chain
+@dataclass
+class DataCollatorForAuto(
+    DataCollatorForTokenClassification,
+    DataCollatorForMultipleChoiceClassification,
+):
+    task: Optional[str] = None
 
-        batch_size = len(features)
-        num_choices = len(features[0]["input_ids"])
-        flattened_features = [
-            [{k: v[i] for k, v in feature.items()} for i in range(num_choices)]
-            for feature in features
-        ]
-        flattened_features = list(chain(*flattened_features))
-        batch = super(DataCollatorForPredict, self).__call__(flattened_features)
-        # Un-flatten
-        batch = {k: v.view(batch_size, num_choices, -1) for k, v in batch.items()}
-        return batch
+    def __call__(self, features):
+        # Call the right data collator based on the task
+        # The MRO looks like below:
+        #    DataCollatorForAuto --> DataCollatorForTokenClassification --> DataCollatorMixin
+        #                        --> DataCollatorForMultipleChoiceClassification --> DataCollatorWithPadding
+        if self.task == MULTICHOICECLASSIFICATION:
+            return super(DataCollatorMixin, self).__call__(
+                features
+            )  # Will call DataCollatorForMultipleChoiceClassification based on MRO (DFS)
+        elif self.task == TOKENCLASSIFICATION:
+            return super().torch_call(
```

Comment:
why `torch_call` here and `__call__` above?

Reply:
Because there are only torch_call and tf_call in DataCollatorForTokenClassification, there is no __call__.